### ISSUE TITLE
Unhandled exception in OSQuery node (tls plugin) over Windows when server is down

### ISSUE THREAD

**Issue by casimanu** [facebook/osquery#4429]
### Description
An unhandled exception is thrown when an OSQuery node starts running using tls plugin, make a request and the server is down.

### What version of `osquery` are you using? 

```
osquery_version = 3.2.4
os_version = 10.0.14393
os_build = 14393
os_platform = windows
```

### What have you already tried?

I created a zip with the files I used for reproducing the problem: [1. unhandled exception in osquery node over windows.zip](https://github.com/facebook/osquery/files/2037606/1.unhandled.exception.in.osquery.node.over.windows.zip)
Along with the files I added a demo video.

Basically I'm running the OSQuery node as:
```
osqueryd.exe --verbose --pidfile tmp/osquery.pid --host_identifier uuid --database_path tmp/osquery.db --enroll_tls_endpoint /enroll --enroll_secret_path "../shared-key.txt" --disable_distributed false --distributed_plugin tls --distributed_interval 10 --allow_unsafe --distributed_tls_read_endpoint /distributed_read --distributed_tls_write_endpoint /distributed_write --tls_dump true --tls_hostname "localhost:8003" --tls_server_certs "../certs/myCA.pem" --config_plugin tls --logger_plugin tls --config_tls_endpoint /config --logger_tls_endpoint /logger --pack_delimiter \
```

Without starting up any server and every time a request is made an unhandled exception is thrown.

Several times I saw the same unhandled exception if the server is up, and at some point, I kill the server but I can not consistently reproduce it in this way.

### What did you expect to see?

The OSQuery node should continue trying to reach the server for enrolling and there should be no unhandled exception.

### What did you see instead?

An unhandled exception

![unhandled-exception](https://user-images.githubusercontent.com/39602188/40521020-c9e6bdb2-5f9f-11e8-9426-da60904de65c.png)

**Comment by fmanco** [comment]
Hi @casimanu can you test this with #4359 applied? It is just a guess at this point but worth a try.

**Comment by casimanu** [comment]
Hi @fmanco. I'm having some problems to set up my OSQuery development environment over Windows. As soon as I get it, I will test the changes in #4359.

**Comment by theopolis** [comment]
I think this is caused by https://github.com/facebook/osquery/issues/4398, check if the 3.2.6 tag fixes it.

**Comment by casimanu** [comment]
@theopolis you are right. The bug was fixed in 3.2.6. I tested this for a while and I never got the unhandled exception.
